### PR TITLE
Superdense coding sample works in Adaptive Profile

### PR DESCRIPTION
**Pull request by DmitryVasilevsky**
Changed Superdense coding example to work with adaptive profile
* Use quantum random bit generator instead of one provided in simulator
* Removed unnecessary check that wasn't compatible with the profile
* Changed return value so that correctness of the algorithm can be verified.
Now it generates QIR in adaptive profile.

CHECK: Not sure if this return type from Main is supported. Need to check.

### CODE REVIEW

**Review by swernli — APPROVED** [review]

File: samples/algorithms/SuperdenseCoding.qs [hunk 1]
```diff
@@ -9,37 +9,39 @@
 /// This Q# program implements superdense coding to send two classical bits of
 /// information.
 namespace Sample {
+    open Microsoft.Quantum.Measurement;
     open Microsoft.Quantum.Diagnostics;
     open Microsoft.Quantum.Random;
```
Comment (billti) [inline samples/algorithms/SuperdenseCoding.qs:14]: I just removed all 3 namespaces locally and it still worked fine. I believe Measurement is automatically included, and there are no Diagnostics or Random calls with these changes.

File: samples/algorithms/SuperdenseCoding.qs [hunk 2]
```diff
@@ -9,37 +9,39 @@
...
         ResetAll([aliceQubit, bobQubit]);
-        return (decodedBit1, decodedBit2);
+        return ((sourceBit1, sourceBit2), (decodedBit1, decodedBit2));
+    }
+
+    // Generates random bit using an auxilliary qubit
+    operation DrawRandomBit() : Bool {
+        use q = Qubit();
+        H(q);
+        MResetZ(q) == One
```
Comment (cesarzc) [inline samples/algorithms/SuperdenseCoding.qs:40]:
```suggestion
        return MResetZ(q) == One;
```

Comment (cesarzc) [reply samples/algorithms/SuperdenseCoding.qs:40]: nit: our samples do not generally use implicit returns (at least currently).

Comment (swernli) [reply samples/algorithms/SuperdenseCoding.qs:40]: aww, I liked having the implicit return in there. But yes, you are right that it's not what we've done so far. When would be the right time to start using that language feature?

**Review by billti — APPROVED** [review]

File: samples/algorithms/SuperdenseCoding.qs (hunk 2 above)
Comment (DmitryVasilevsky) [reply samples/algorithms/SuperdenseCoding.qs]: Added return.

**Pull request merged**